### PR TITLE
QUARKUS-2388 Promote Spring Data REST from TP to full support

### PR DESCRIPTION
**Pull request by jsmrcka**
https://issues.redhat.com/browse/QUARKUS-2388

Promote Spring Data REST from TP to full support

Documentation: https://quarkus.io/version/main/guides/spring-data-rest

### CODE REVIEW

File: QUARKUS-2388.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- PagingAndSortingRepository
+
+[Quarkus QE test suite](https://github.com/quarkus-qe/quarkus-test-suite/tree/main/spring/spring-data)
+- CrudRepository
+- PagingAndSortingRepository
+- @RepositoryRestResource
+- Use of just one data type: `application/json`
+- Native and OpenShift coverage
+
+## Automated test development
```
Comment (rsvoboda) [inline QUARKUS-2388.md:38]: What's the planned effort to for new TD?

File: QUARKUS-2388.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Quarkus QE test suite - module `spring/spring-data`:
+- Add JpaRepository test scenario.
+- Include alternative data type (`application/hal+json`) tests in the scenario.
+
```
Comment (rsvoboda) [inline QUARKUS-2388.md:42]: I would include a "negative" test(s) - findById/deleteById invoked with invalid id, save of corrupted entity, etc ...

Comment (jsmrcka) [reply QUARKUS-2388.md:44]: Added.

File: QUARKUS-2388.md (hunk 1 above)
Comment (jsmrcka) [reply QUARKUS-2388.md:39]: ???

Comment (rsvoboda) [reply QUARKUS-2388.md:39]: How long will it take to implement planned tests?

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**